app/code/community/Payone/Core/Model/Service/TransactionStatus/Execute.php

```diff
@@ -198,91 +288,77 @@ public function execute(Payone_Core_Model_Domain_Protocol_TransactionStatus $tra
         $this->getApp()->loadAreaPart(Mage_Core_Model_App_Area::AREA_FRONTEND, Mage_Core_Model_App_Area::PART_TRANSLATE);
         $this->getApp()->loadAreaPart(Mage_Core_Model_App_Area::AREA_FRONTEND, Mage_Core_Model_App_Area::PART_DESIGN);
 
-        // Track current transaction status as processed (no matter what happens next).
-        $this->processed[] = $transactionStatus;
+        // Set status to running to kind of lock this TX status.
+        // TODO: There is a problem if the processing fails and the TX status remains in running status for some reason.
+        // The TX status will not be processed again because of the dangling running status.
+        // https://github.com/PAYONE-GmbH/magento-1/issues/428
+        $txStatus->setStatusRunning();
+        $txStatus->save();
```

Comment:
Based on other feedback read in issues posts, there could be some failure happening already during the "$txStatus->save()" step, which are locking some transaction in Running state without being processed even once.
Moving those three lines [setRunnning(), save(), logMessage()] downward inside the try/catch block could help to handle smoother the "early failure" cases.
It still has to be checked what will happen to those transaction, and see if it matches with the process used for other failure cases.
(example of the change in : https://github.com/PAYONE-GmbH/magento-1/pull/440)